FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs

```diff
@@ -132,9 +146,16 @@ await foreach (FhirToDataLakeProcessingJobInputData input in inputs.WithCancella
                     _result.CreatedJobCount++;
                     _result.RunningJobIds.Add(newJobId);
 
-                    // if enqueue successfully while fails to report result, will re-enqueue and return the existing jobInfo
+                    foreach (var item in input.SplitParameters)
+                    {
+                        _result.SubmittedResourceTimestamps[item.Key] = item.Value.DataEndTime;
+                    }
+
+                    _result.SubmittingProcessingJob = null;
                     progress.Report(JsonConvert.SerializeObject(_result));
 
+                    _jobStatus = await UpdateJobStatusWithResultAsync(cancellationToken);
```

Comment:
seems we serialize result twice for report result and update job status. can we merge them?

Reply:
Refined it, thanks.